cloud/run/hooks.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""User-defined code that is hooked into the translation workflow."""
```

Comment:
I think a bit bigger example is needed here. 

Few things that customers are doing in pre-post processing where "path" is the context.

Temp tables
PRE: find temp tables, change them to regular ones
POST: add drop

Insert into external
PRE: change into regular table
POST: add export job to GCS

all of those requires storing contextual information in this file. 
Assuming this is multithreaded, is it user responsibility to worry about locking/thread safe access?

Reply:
Hi @stankiewicz, thanks for the feedback!

> I think a bit bigger example is needed here.

The REAMDE is still a TODO on this WIP PR and will include details on the hooks.py file. I will also add a TODO to include inline examples in the file itself.

> Temp tables
> PRE: find temp tables, change them to regular ones
> POST: add drop

I thing there might be an internal bug to add this to the service that someone is already working on. Will double check.

> Insert into external
> PRE: change into regular table
> POST: add export job to GCS

Perhaps there should be an internal bug for this as well?

However, you are correct, these are normally use cases we used to address in Groovy with the binary and should be supported here in Python as well.

> Assuming this is multithreaded, is it user responsibility to worry about locking/thread safe access?

@xiashang0624 convinced me that single-threaded should be the default and you make another good point as to why this should be the case. I have already modified this PR to make single-threaded the default. Will be sure to add a note/warning in the README about hooks.py when multithreading is enabled.